Fix panache repository operations failure with multiple persistence units

machi1990 opened the pull request:
Fixes https://github.com/quarkusio/quarkus/issues/11842

machi1990 committed:
fix(panache): repository operations failure with multiple persistence units in 1.8.0.cr1

Fixes https://github.com/quarkusio/quarkus/issues/11842

gsmet commented on extensions/panache/hibernate-orm-panache/deployment/src/main/java/io/quarkus/hibernate/orm/panache/deployment/PanacheHibernateResourceProcessor.java:
```diff
@@ -186,6 +197,10 @@ void build(CombinedIndexBuildItem index,
+
+        panacheEntities.addAll(modelClasses);
+
+        recordPanacheEntityPersistenceUnits(recorder, jpaModelPersistenceUnitMapping, panacheEntities);
```
Hmmm. Do we support using both approaches correctly?

I haven't checked thoroughly but my bet would be it won't work.

gsmet replied:
I'm not sure I was clear but I meant: mixing Panache entities and Panache repositories approaches.

machi1990 replied:
I am not sure it is clear enough. You mean a `PanacheEntity` cannot have the corresponding Panache repository? 
If so, I think it is very possible. I once used that pattern here: 
- [TodoRepository.java](https://github.com/machi1990/todo-api-with-quarkus-oidc/blob/master/machi-todo-api/src/main/java/org/machi/api/repository/TodoRepository.java) 
- [Todo.java](https://github.com/machi1990/todo-api-with-quarkus-oidc/blob/master/machi-todo-api/src/main/java/org/machi/api/entity/Todo.java)

gsmet replied:
No I mean: what happens if you have:
- Entity1 which is an entity using the Active Record pattern and inheriting from PanacheEntity
- Entity2 which uses the Panache repository pattern
all in the same project.

machi1990 replied:
This should work too. 

It's what I've in my test case I just did not extend it enough: let me add it so that we can confirm that it works.

machi1990 replied:
@gsmet I've added this test case in a separate commit https://github.com/quarkusio/quarkus/pull/11852/commits/789e6ed9d0b20461d3258c3357e107b401353e1c

gsmet replied:
OK, perfect if we have a test!

machi1990 committed:
test(panache): add a test case to show that we can mix the repository and active record pattern

Follows up https://github.com/quarkusio/quarkus/pull/11852#discussion_r482944985
